ArduCopter/AP_Arming.cpp

```diff
@@ -437,7 +437,7 @@ bool AP_Arming_Copter::gps_checks(bool display_failure)
 
     // warn about hdop separately - to prevent user confusion with no gps lock
     if (copter.gps.get_hdop() > copter.g.gps_hdop_good) {
-        check_failed(ARMING_CHECK_GPS, display_failure, "High GPS HDOP");
+        check_failed(ARMING_CHECK_GPS, display_failure, "High GPS HDOP(%.2f>%.2f)", copter.gps.get_hdop()/100.0f, uint16_t(copter.g.gps_hdop_good)/100.0f);
```

Comment:
uint16_t(copter.g.gps_hdop_good)/100.0f) should be (double)(copter.g.gps_hdop_good/100.0f) and copter.gps.get_hdop() should be casted to double too !

Reply:
@khancyr san. I'll change.